### PR TITLE
EVG-16350 Add MergedFrom to commit queue merge patches

### PR DESCRIPTION
**Pull request by bynn**
[EVG-16350](https://jira.mongodb.org/browse/EVG-16350)

### Description 
Add MorgedFrom field so that when you enqueue patch from cli or ui, it will contain the patch id of the original patch it was based off

### Testing 
unit test

tested on spruce 
https://evergreen-staging.corp.mongodb.com/rest/v2/patches/623b949e97b1d310aeb0b06d

on cli 
https://evergreen-staging.corp.mongodb.com/rest/v2/patches/623cbaa19dbe3216e5c60c76

### CODE REVIEW

File: model/patch/patch.go [hunk 1]
```diff
@@ -172,6 +172,8 @@ type Patch struct {
 	// MergeStatus is only used in gitServePatch to send the status of this
 	// patch on the commit queue to the agent
 	MergeStatus string `json:"merge_status"`
+	// MergedFrom is only used when a merge patch is created from an existing patch.
```
Comment (ablack12) [inline model/patch/patch.go:175]: Can this specify that the value is set to the patch ID of the existing patch?

**Review by ablack12 — COMMENTED** [review]
Were you able to test both the CLI and UI methods of merging?

File: model/patch/patch.go (hunk 1 above)
Comment (bynn) [reply model/patch/patch.go]: done

File: model/patch/patch.go [hunk 2]
```diff
@@ -172,6 +172,9 @@ type Patch struct {
 	// MergeStatus is only used in gitServePatch to send the status of this
 	// patch on the commit queue to the agent
 	MergeStatus string `json:"merge_status"`
+	// MergedFrom is populated with patch id of the existing patch
```
Comment (ablack12) [inline model/patch/patch.go:175]: nit: "with the"

Comment (ablack12) [reply model/patch/patch.go:175]: and maybe also an ", if applicable." at the end since this isn't typically populated

**Review by ablack12 — APPROVED** [review]

**Pull request merged**